documentation/12-configuration/README.md

```diff
@@ -0,0 +1,270 @@
+Title: Configuration
+----
+Text:
+
+ContainerPilot expects a configuration file with details about what services it should register, how to check their health, and what to do at startup and shutdown, among others.
+
+### Specifying the configuration file
+
+There are a few ways to specify the ContainerPilot configuration location.
+
+1. An environment variable set in the `docker run...` string, in a Docker Compose manifest, in the Dockerfile, or elsewhere
+2. As an argument to ContainerPilot, perhaps in the the `docker run...` string, or in in the `CMD` or `ENTRYPOINT` of the Dockerfile
+3. As a JSON string passed to ContainerPilot in any of the locations named above
+
+#### Examples: specifying the configuration file path
+
+```bash
+# configure via passing a file argument
+$ containerpilot -config file:///opt/containerpilot/app.json myapp --args --for --my --app
+
+# configure via environment variable
+$ export CONTAINERPILOT=file:///opt/containerpilot/app.json
+$ containerpilot myapp --args --for --my --app
+```
+
+Many of the Autopilot Pattern implementations specify the configuration file path via an environment variable in the Dockerfile. [See the Nginx implementation for an example](https://github.com/autopilotpattern/nginx/blob/master/Dockerfile).
+
+### Configuration file
+
+The format of the JSON file configuration is as follows:
+
+```json
+{
+  "consul": "consul:8500",
+  "preStart": "/opt/containerpilot/preStart-script.sh {{.ENV_VAR_NAME}}",
```

Comment:
Do we want to update this example config to use `/use/local/bin/some-script.sh` and `/etc/containerpilot.json` as we've more-or-less standardized on elsewhere in our Autopilot Pattern examples?